File: 1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one.py
class Solution:
  def numSteps(self, s: str) -> int:
    ans = 0
    n = int(s, 2)
    while n > 1:
      ans += 1
      if n & 1:
        n += 1
      else:
        n >>= 1
    return ans
        

File: 1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one.js
const numSteps = (s) => {
  let ans = 0;
  while (s !== '1') {
    if (s[s.length - 1] === '1') {
      const i = s.lastIndexOf('0');
      if (i === -1) {
        s = '1' + ''.padStart(s.length, '0')
      } else {
        s = s.substring(0, i) + '1' + ''.padStart(s.length - (i + 1), '0')
      }
    } else {
      s = s.substring(0, s.length - 1);
    }
    ans++
  }
  return ans;
};